Sources/GroundState/vmc_sampling.cc

```diff
@@ -0,0 +1,165 @@
+#include "vmc_sampling.hpp"
```

Comment:
I guess that vmc_sampling and the more general routines should sit at different level than `GroundState`. Maybe `Sources/MonteCarlo/machine_sampling.cc` or something similar? 
In this way it is also more obvious that these routines can be reused in other type of learning algorithms, beyond ground state

Reply:
Even `Sources/Sampler/machine_sampling.cc` would be a good option, maybe even better than creating a new folder

Reply:
Moved the code to `Sampler/`.